### PR TITLE
Сustom options

### PR DESCRIPTION
**Pull request by exileed**
Added ->setOption(); method for custom options.

Thanks.

### CODE REVIEW

File: src/OptionMapping.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Spatie\Docker;
+
+class OptionMapping
```
Comment (freekmurze) [inline src/OptionMapping.php:5]: Rename this to `Option`

File: src/OptionMapping.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{
+
+    private string $optionName;
```
Comment (freekmurze) [inline src/OptionMapping.php:8]: Change `private` to `protected`

File: src/OptionMapping.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * @var int|string
+     */
+    private $optionValue;
+
+    /**
+     * ArgumentMapping constructor.
```
Comment (freekmurze) [inline src/OptionMapping.php:16]: Remove this unneeded comment

**Review by freekmurze — CHANGES_REQUESTED** [review]
Could you also add a new method `setOptions` which accepts an array? Also update the readme and tests for that one.

Thanks for your work on this.

**Comment by exileed** [comment]
> 
> 
> Could you also add a new method `setOptions` which accepts an array? Also update the readme and tests for that one.
> 
> Thanks for your work on this.

Done.

File: src/DockerContainer.php [hunk 4]
```diff
@@ -109,6 +112,33 @@ public function setEnvironmentVariable(string $envName, string $envValue): self
...
+     */
+    public function setOption(string $optionName, $valueName): self
+    {
+        $this->options[] = new Option($optionName, $valueName);
+
+        return $this;
+    }
+
+    /**
+     * @param array $options
```
Comment (freekmurze) [inline src/DockerContainer.php:129]: This comment doesn't bring any value, remove it

File: src/Option.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Spatie\Docker;
+
+class Option
+{
+
+    protected string $optionName;
+
+    /**
+     * @var int|string
```
Comment (freekmurze) [inline src/Option.php:11]: For consistency use a one-line docblock

**Review by freekmurze — CHANGES_REQUESTED** [review]
Nearly there!

Could you also document the `setOptions` method in the readme?

**Pull request closed without merging**